File: kraken/lib/src/element/bounding_client_rect.dart
/*
 * Copyright (C) 2020-present Alibaba Inc. All rights reserved.
 * Author: Kraken Team.
 */

class BoundingClientRect {
  final double x;
  final double y;
  final double width;
  final double height;
  final double top;
  final double left;
  final double right;
  final double bottom;

  const BoundingClientRect(
      {this.x = 0.0,
      this.y = 0.0,
      this.width = 0.0,
      this.height = 0.0,
      this.top = 0.0,
      this.left = 0.0,
      this.right = 0.0,
      this.bottom = 0.0});

  String toJSON() {
    return '{"x": $x, "y": $y, "width": $width, "height": $height, "top": $top, "left": $left, "right": $right, "bottom": $bottom}';
  }

  String toString() {
    return 'BoundingClientRect(${toJSON()})';
  }
}

File: bridge/core/dom/legacy/bounding_client_rect.cc
/*
 * Copyright (C) 2021-present The Kraken authors. All rights reserved.
 */

#include "bounding_client_rect.h"
#include "core/executing_context.h"

namespace kraken {

BoundingClientRect* BoundingClientRect::Create(ExecutingContext* context,
                                               NativeBoundingClientRect* native_bounding_client_rect) {
  return MakeGarbageCollected<BoundingClientRect>(context, native_bounding_client_rect);
}

BoundingClientRect::BoundingClientRect(ExecutingContext* context, NativeBoundingClientRect* nativeBoundingClientRect)
    : ScriptWrappable(context->ctx()),
      x_(nativeBoundingClientRect->x),
      y_(nativeBoundingClientRect->y),
      width_(nativeBoundingClientRect->width),
      height_(nativeBoundingClientRect->height),
      top_(nativeBoundingClientRect->top),
      right_(nativeBoundingClientRect->right),
      left_(nativeBoundingClientRect->left),
      bottom_(nativeBoundingClientRect->bottom) {}

void BoundingClientRect::Trace(GCVisitor* visitor) const {}

}  // namespace kraken
